packages/core-data/src/selectors.ts

```diff
@@ -1222,6 +1236,28 @@ export function getBlockPatternCategories( state: State ): Array< any > {
 	return state.blockPatternCategories;
 }
 
+/**
+ * Retrieve the registered user pattern categories.
+ *
+ * @param state Data state.
+ *
+ * @return User patterns category array and map keyed by id.
+ */
+
+export function getUserPatternCategories(
+	state: State
+): UserPatternCategories {
+	const patternCategoriesMap = new Map< number, UserPatternCategory >();
+	state.userPatternCategories?.forEach(
+		( userCategory: UserPatternCategory ) =>
+			patternCategoriesMap.set( userCategory.id, userCategory )
+	);
+	return {
+		patternCategories: state.userPatternCategories,
+		patternCategoriesMap,
+	};
+}
```

Comment:
This needs to be memorized. The `useSelect` noticed from this screenshot is caused by this selector - https://github.com/WordPress/gutenberg/pull/53835#issuecomment-1720669088.

Aside from this, is there a reason we're creating a map based on the state inside a selector? The task is better suited for consumers, in my opinion.

Reply:
It doesn't look like this selector was causing that screenshot error as that error was there on trunk prior to this merge - we will look at that separately. 

This selector was causing the useBlockEditorSettings useMemo return value to be rerun with each call though, so have [moved the map into the consumers as suggested](https://github.com/WordPress/gutenberg/pull/54576). While investigating this I also discovered another change which was causing this useMemo to be reevaluated with every call of the hook. This has been [fixed here](https://github.com/WordPress/gutenberg/pull/54580).

Reply:
I meant the warning, visible at the top of the error and pointing to the `useBlockEditorSettings`.

## Screenshot
![CleanShot 2023-09-19 at 09 46 31](https://github.com/WordPress/gutenberg/assets/240569/631952ff-6a39-4fc7-a041-41ab4ed1becb)